ucsb-cs156-s24/proj-gauchoride-s24-5pm-8
# XS - EPIC: Improve usability of the DriverAvailabilities Create Page

## Dokku: https://gauchoride-qa.dokku-16.cs.ucsb.edu

(Deployed on gauchoride-qa as gauchoride-xinyao-dev is taken for PR #23 concurrently)

The PR modifies the `DriverAvailabilitiesForm` page so a driver can understand what each field in the `Create Availability Form` means and submit a correct availability without running into issues. Furthermore, the PR removed the necessity for drivers to input their `driver ID` each time they submit an availability form where the `driver ID` is automatically recorded. In this case, most number of changed lines is due to a combination of having to touch related documentation to the `driverId` field and adding new tests. 

### Testing Instructions:
1. Click `Admin` in the nav bar and go the `Users` page to toggle `Driver` under your name. 
2. Once `Driver` is being toggled, click `Shifts` in the nav bar and go to the `Availability` page. 
3. From there you can see a button called `Create Driver Availability` and that is where you can fill out a driver availability form and test out the functionality of this PR. 

### Acceptance Criteria:

- [ ] The form labels are improved so that each word are spaced out and is easily readable to the user. 
- [ ] The bad input and missing input error messages are added for `Start Time`, `End Time`, and `Day of the Week`. 
- [ ] There are example inputs on each form field which helps the user to know what to put there. 
- [ ] `DriverId` is removed from all aspects of DriverAvailability, including `DriverAvailabilityController.java`, `DriverAvailabilityForm.js`, `DriverAvailabilityEditPage.js`, and `DriverAvailabilityCreatePage.js`. 
- [ ] User is able to input additional notes in the `notes` field if there are anything additional to say. 

### Frontend Preview:
#### Before:
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/448b6577-033f-4a07-a03d-13bd7d9b69d6)
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/dbeed783-7f61-4175-83b0-cb5412dd81f8)
![image](https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/3cea36aa-052a-4ca1-a75c-642a2a871ceb)

#### After: 
<img width="1439" alt="Screenshot 2024-05-25 at 8 36 45 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/198f261f-fb88-41bd-92aa-b530932cfb53">
<img width="1440" alt="Screenshot 2024-05-25 at 8 36 52 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/07a6a344-e4d9-44d6-8a49-f312a1dc4a61">
 
**Example of driver's availability created without additional notes (id:8):** 
<img width="1440" alt="Screenshot 2024-05-25 at 8 37 08 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/d3cda159-7bce-4e8f-928a-725381c1d64c">

**Example of driver's availability created with additional notes (id:9):** 
<img width="1440" alt="Screenshot 2024-05-25 at 8 37 39 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-8/assets/42160992/1f42e3d1-502b-4bf9-aad6-fc8dd1942e8e">
Closes #11
